modules/genomic_browser/php/endpoints/snpbrowser.class.inc

```diff
@@ -0,0 +1,474 @@
+                'Markers'             => parent::formElementObjectCreator(
+                    'Markers',
+                    'text',
+                    'Markers',
+                    true,
+                    null
+                ),
+                'Validation_Method'   => parent::formElementObjectCreator(
+                    'Validation_Method',
+                    'text',
+                    'Validation Method',
+                    true,
+                    null
+                ),
+                'Validated'           => parent::formElementObjectCreator(
+                    'Validated',
+                    'select',
+                    'Validated',
+                    true,
+                    array(
+                     '0' => '0',
+                     '1' => '1',
+                    )
+                ),
+                'Function_Prediction' => parent::formElementObjectCreator(
+                    'Function_Prediction',
+                    'select',
+                    'Function Prediction',
+                    false,
+                    array(
+                     'exonic'      => 'exonic',
+                     'ncRNAexonic' => 'ncRNAexonic',
+                     'splicing'    => 'splicing',
+                     'UTR3'        => 'UTR3',
+                     'UTR5'        => 'UTR5',
+                    )
+                ),
+                'Damaging'            => parent::formElementObjectCreator(
+                    'Damaging',
+                    'select',
+                    'Damaging',
+                    false,
+                    array(
+                     'D'  => 'D',
+                     'NA' => 'NA',
+                    )
+                ),
+                'Genotype_Quality'    => parent::formElementObjectCreator(
+                    'Genotype_Quality',
+                    'text',
+                    'Genotype Quality',
+                    true,
+                    null
+                ),
+                'Exonic_Function'     => parent::formElementObjectCreator(
+                    'Exonic_Function',
+                    'text',
+                    'Exonic Function',
+                    false,
+                    null
+                ),
+                'genomic_range'       => parent::formElementObjectCreator(
+                    'genomic_range',
+                    'text',
+                    'Genomic Range',
+                    true,
+                    null
+                ),
+               );
+    }
+
+    /**
+     * Permission verification.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
```

Comment:
same here. I'm not going to keep marking them but please apply the same change in all files.